library/kani/src/shadow.rs

```diff
@@ -81,3 +81,117 @@ impl<T: Copy> ShadowMem<T> {
         self.mem[obj][offset] = val;
     }
 }
+
```

Comment:
Moved memory initialization checking code into a separate module. 

Talked about the unit tests offline. They are hard to implement because all the calls from the added functions lead to `kani_intrinsic`. In theory, we could shim the intrinsics, but decided not to do it right now